### PR TITLE
Graphql union types for queries

### PR DESCRIPTION
**Pull request by nicolas-scheers-lemon**
## Description
<Describe the purpose of this pull request>

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly

### CODE REVIEW

**Comment by hendrik-depauw-lemon** [comment]
Hey @javiertoledo, have you had the time yet to take a look at this PR? Thanks in advance!

**Comment by hendrik-depauw-lemon** [comment]
Hey @javiertoledo, can I bump this PR? 🙂

File: packages/framework-integration-tests/src/read-models/book-read-model.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+@ReadModel({
+  authorize: 'all',
+})
+export class BookReadModel {
+  public constructor(readonly id: UUID, readonly title: string, readonly pages: number) {}
+
+  @Projects(Book, 'id')
+  public static updateBook(book: Book, oldBook?: BookReadModel): ProjectionResult<BookReadModel> {
+    // This method calls are here to ensure they work. More info: https://github.com/boostercloud/booster/issues/797
+    book.getId()
```
Comment (davidverdu) [inline packages/framework-integration-tests/src/read-models/book-read-model.ts:14]: I think this workaround is not necessary. Did you have issues when you didn't add this line of code?

**Review by davidverdu — COMMENTED** [review]
@nicolas-scheers-lemon Great work! I left two comments. Please let me know when they are addressed, so that we can approve it

**Comment by davidverdu** [comment]
nonetheless @nicolas-scheers-lemon , you also need to add the changeset file. Please run `rush change` and commit the changeset file

**Comment by nicolas-scheers-lemon** [comment]
@davidverdu I addressed your comments and added the changeset file

**Review by MarcAstr0 — APPROVED** [review]

**Review by davidverdu — APPROVED** [review]

**Comment by davidverdu** [comment]
Hi @nicolas-scheers-lemon @hendrik-depauw-lemon the PR is approved. However, you need to solve the conflicts, for example, by rebasing. Could you do that?

**Comment by nicolas-scheers-lemon** [comment]
@davidverdu I solved the conflict

**Pull request merged**